### DISCUSSION TITLE
Are DuckDB operations order preserving?

### DISCUSSION THREAD

**Discussion by marsupialtail** [duckdb/duckdb#7953] [General]
Specifically, filters, asof joins, and range joins

**Comment by Alex-Monahan** [comment]
I believe only filters are order preserving in your list. If you force only 1 thread, more operations may preserve order, but I am not certain!

**Reply by marsupialtail** [reply]
Are filters order-preserving even with multiple threads?

**Reply by Alex-Monahan** [reply]
I believe so

**Reply by Tishj** [reply]
Yes, that's what we use batch indexes for :)

In case you'e interested, here's the PR that introduced batch indexes
<https://github.com/duckdb/duckdb/pull/3700>

**Reply by marsupialtail** [reply]
FWIW I am not sure if that is true. 

```
print(crimes["TIMESTAMP_LONG"].is_sorted()) # True
crimes_a = crimes.to_arrow()
polars.from_arrow(con.execute("select * from crimes_a where primary_category_id = 1 OR primary_category_id = 24 OR primary_category_id = 27").arrow())["TIMESTAMP_LONG"].is_sorted() # False
```

**Reply by arjenpdevries** [reply]
I think it's unlikely that they would.

Better add "order by" if you need a specific ordering!

**Reply by Tishj** [reply]
If no order is given then there is no order to preserve.
